guides/k8-akka-cluster/src/main/paradox/index.md

```diff
@@ -0,0 +1,465 @@
+          {
+            "name": "myapp",
+            "image": "mygroup/myapp",
+            "imagePullPolicy": "Never",
+            "ports": [
+              {
+                "containerPort": 2551,
+                "name": "akka-remote"
+              }
+            ],
+            "resources": {
+              "limits": {
+                "cpu": "250m",
+                "memory": "384Mi"
+              },
+              "requests": {
+                "cpu": "250m",
+                "memory": "384Mi"
+              }
+            },
+            "env": [
+              {
+                "name": "AKKA_ACTOR_SYSTEM_NAME",
+                "value": "myapp-actor-system"
+              },
+              {
+                "name": "AKKA_REMOTING_BIND_PORT",
+                "value": "2551"
+              },
+              {
+                "name": "AKKA_REMOTING_BIND_HOST",
+                "valueFrom": {
+                  "fieldRef": {
+                    "fieldPath": "metadata.name"
+                  }
+                }
+              },
+              {
+                "name": "AKKA_SEED_NODE_PORT",
+                "value": "2551"
+              },
+              {
+                "name": "AKKA_SEED_NODE_HOST",
+                "value": "myapp-0"
+              }
+            ],
+            "readinessProbe": {
+              "tcpSocket": {
+                "port": 2551
+              },
+              "initialDelaySeconds": 30,
+              "timeoutSeconds": 30
+            }
+          }
+        ]
+      }
+    }
+  }
+}
+EOF
+```
+
+The StatefulSet for the application has `myapp` as the name as declared by `spec.serviceName`.
+
+The StatefulSet has `1` replica. You may adjust this number to the number of instances you desire. Alternatively, you can deploy the with `1` replica to begin with, and once the first service is started you may adjust this number by modifying the StatefulSet:
+
+```bash
+kubectl scale statefulsets myapp --replicas=3
+```
+
+The Pods for the StatefulSet will be initialized with the image `mygroup/myapp`. Adjust this image name to match the actual image published by the SBT or Maven build. The `imagePullPolicy` is set to `Never` to ensure only image published to the Docker registry is used. If the image doesn't exist in the Docker registry, the StatefulSet creation will fail with an error.
+
+Each Pod in the StatefulSet will expose port `2551` as declared by the `containerPort` named `akka-remote`.
+
+Please adjust the CPU and memory resources to match the requirements of the application.
+
+The `env` property of the StatefulSet contains list of environment variable to be passed into the application when it starts. These environment variables will be used to populate the system properties to bootstrap the Akka cluster required by the application. Note that the environment variable `AKKA_SEED_NODE_HOST` is set to `myapp-0`, which means that `myapp-0` is used as the seed node to establish the Akka cluster.
```

Comment:
I think we'll be relying on the fact that the pod names generated by `StatefulSet` has orderly numbered suffix, i.e. `myapp-0`, `myapp-1`, `myapp-2`, and so on.

So given a pod called `myapp-2`, `StatefulSet` guarantees there would be `myapp-0` and `myapp-1` which is then we can use as seed nodes.

I haven't put this into the guide since that will mean doing all these parsing in the application startup script, which has flow on effect to both SBT and Maven builds.

SBT builds relies on SBT native packager which generates its own application startup script by default, and therefore we need to override parts of the default script. This is easily done from `build.sbt`, but we haven't figured out how to do this outside of `build.sbt`.

With Maven, we need to generate a startup script as part of the `docker:package` and set it correctly as the `<entryPoint>` in the plugin config since it's not possible to perform this parsing within the `<entryPoint>`.

These won't be a problem with tooling, i.e. `akka-kubernetes-lib` equivalent of `akka-conductr-lib` - since it will allow us to create the typesafe config object, including the seed nodes configuration from these environment variable.

We're definitely have tooling planned soon after the docs, and that's why I haven't put it in the docs.